Add salt state module to manage kibana objects

gthvn1 opened the pull request.

Commit:
Manage Kibana objects

This patch adds a salt state module to manage Kibana objects. It also
adds the client to install these objects.

thouveng commented:
!!! Don't merge: work is in progress !!!

elemoine commented on _states/kibana_object.py:
```diff
@@ -0,0 +1,101 @@
+
+    profile = __salt__['config.option']('kibana')
+
+    url = profile.get('kibana_url', None)
```
You don't need the second parameter (`None`).

gthvn1 replied:
Right it is the default.

Commit:
Add Kibana StackLight objects

This patch adds the following Kibana objects:
  dashboard:
  - dashboard_logs
  - dashboard_notifications
  index-pattern:
  - log-*
  - notification-*
  search:
  - search-logs
  - search-notifications
  visualization:
  - LOG-MESSAGES-OVER-TIME-PER-SEVERITY
  - LOG-MESSAGES-OVER-TIME-PER-SOURCE
  - NUMBER-OF-LOG-MESSAGES-PER-ROLE
  - NUMBER-OF-LOG-MESSAGES-PER-SEVERITY
  - TOP-10-HOSTS
  - TOP-10-PROGRAMS
  - TOP-10-SOURCES
  - NOTIFICATIONS-OVER-TIME-PER-SOURCE
  - NOTIFICATIONS-OVER-TIME-PER-SEVERITY
  - EVENT-TYPE-BREAKDOWN
  - SOURCE-BREAKDOWN
  - HOST-BREAKDOWN
  - NOTIFICATIONS-PER-SEVERITY

gthvn1 commented:
I added the possibility to delete an object. I also manage an update.
